components/products/ProductCardList.tsx

```diff
@@ -1,12 +1,27 @@
-import React from "react";
+"use client";
+
+import React, { useEffect, useState } from "react";
 import ProductCard from "./ProductCard";
 import { Product } from "../../entities/Product";
+import { FetchProducts } from "@/app/apis/products";
 
 interface ProductCardListProps {
-  products: Product[];
+  products?: Product[];
 }
 
-const ProductCardList: React.FC<ProductCardListProps> = ({ products }) => {
+const ProductCardList: React.FC<ProductCardListProps> = () => {
+  const [products, setProducts] = useState<Product[]>([]);
+
+  useEffect(() => {
+    const apiUrl = process.env.NEXT_PUBLIC_API_URL;
+
+    if (!apiUrl) {
+      console.error("API URL is not defined");
+      return;
+    }
+    FetchProducts(apiUrl).then(setProducts).catch(console.error);
```

Comment:
Two thoughts here: 
1. What do you think about using SSR here and getting the data from the server side? 
2. If we stick to Client side request and CSR, I highly recommend using libraries like Tanstack Query to handle the asynchronous calls.

Reply:
Change to use SSR